lib/queueClient.ts

```diff
@@ -0,0 +1,209 @@
+const debug = debugModule("azure:service-bus:queue-client");
+
+/**
+ * The mode in which messages should be received
+ */
+export enum ReceiveMode {
+  /**
+   * Peek the message and lock it until it is settled or times out.
+   * @type {Number}
+   */
+  peekLock = 1,
+
+  /**
+   * Remove the message from the service bus upon delivery.
+   * @type {Number}
+   */
+  receiveAndDelete = 2
+}
+
+/**
+ * Describes the options that can be provided while creating the QueueClient.
+ * @interface QueueClientOptions
+ */
+export interface QueueClientOptions {
+  /**
+   * @property {number} [receiveMode] The mode in which messages should be received.
+   * Default: ReceiveMode.peekLock
+   */
+  receiveMode?: ReceiveMode;
+  /**
+   * @property {number} [maxConcurrentCalls] The maximum number of messages that should be
+   * processed concurrently while in peek lock mode. Once this limit has been reached, more
+   * messages will not be received until messages currently being processed have been settled.
+   * Default: 1
+   */
+  maxConcurrentCalls?: number;
+}
+
+export class QueueClient extends Client {
+  /**
+   * @property {number} receiveMode The mode in which messages should be received.
+   * Default: ReceiveMode.peekLock
+   */
+  receiveMode: ReceiveMode;
+  /**
+   * @property {number} maxConcurrentCalls The maximum number of messages that should be
+   * processed concurrently while in peek lock mode. Once this limit has been reached, more
+   * messages will not be received until messages currently being processed have been settled.
+   */
+  maxConcurrentCalls: number;
+
+  /**
+   * Instantiates a client pointing to the ServiceBus Queue given by this configuration.
+   *
+   * @constructor
+   * @param {string} name The Queue name.
+   * @param {ConnectionContext} context The connection context to create the QueueClient.
+   * @param {QueueClientOptions} [options] The QueueClient options.
+   */
+  constructor(name: string, context: ConnectionContext, options?: QueueClientOptions) {
+    super(name, context);
+    if (!options) options = {};
+    this.receiveMode = options.receiveMode || ReceiveMode.peekLock;
+    this.maxConcurrentCalls = options.maxConcurrentCalls != undefined ? options.maxConcurrentCalls : 1;
+  }
+
+  /**
+   * Closes the AMQP connection to the ServiceBus Queue for this client,
+   * returning a promise that will be resolved when disconnection is completed.
+   * @returns {Promise<any>}
+   */
+  async close(): Promise<any> {
+    try {
+      if (this._context.namespace.connection && this._context.namespace.connection.isOpen()) {
+        // Close the sender.
+        if (this._context.sender) {
+          await this._context.sender.close();
+        }
+        // Close the receiver.
+        if (this._context.streamingReceiver) {
+          await this._context.streamingReceiver.close();
+        }
+        // Close the management session
+        await this._context.managementSession!.close();
```

Comment:
is this management session shared between all clients on the namespace? if so, won't closing it break the other clients?